### PR TITLE
check arithmetic operation overflow

### PR DESCRIPTION
**Pull request by siddontang**
+ support +, - and *.
+ support /,  DIV and MOD later.

### CODE REVIEW

File: util/types/overflow.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		res, err = MulUint64(uint64(a), uint64(-b))
+	}
+
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	if negative {
+		// negative result
+		if res > math.MaxInt64+1 {
```
Comment (qiuyesuifeng) [inline util/types/overflow.go]: Here why not using `res > math.MaxInt64`?

**Comment by coocood** [comment]
LGTM

File: util/types/overflow.go (hunk 1 above)
Comment (qiuyesuifeng) [inline util/types/overflow.go]: It is ok here, for `| math.MaxInt64 | + 1 = | math.MinInt64 | `.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by zimulala** [comment]
arthimetic -> arithmetic?

**Pull request merged**